tests/test_plugins.py

```diff
@@ -211,3 +212,264 @@ def test_imds(
+    {
+        "KUBERNETES_PORT": "tests",
+    },
+)
+@pytest.mark.parametrize("copy_files", [[LS_PATH]], indirect=True)
+@pytest.mark.parametrize("tmp_file", [{"path": "/tmp/vendor"}], indirect=True)
+def test_imds_eks(
+    copy_files: list[Path],
+    chalk: Chalk,
+    tmp_file: IO,
+    server_imds: str,
+):
+    # make imds plugin think we are running in EC2
+    with tmp_file as fid:
+        fid.write(b"Amazon")
+    bin_path = copy_files[0]
+    insert = chalk.insert(bin_path, config=CONFIGS / "imds.conf")
+    assert insert.report.contains(
+        {
+            "_OP_CLOUD_PROVIDER": "aws",
+            "_OP_CLOUD_PROVIDER_SERVICE_TYPE": "aws_eks",
+        }
+    )
+
+
+@pytest.mark.parametrize("copy_files", [[LS_PATH]], indirect=True)
+@pytest.mark.parametrize("tmp_file", [{"path": "/tmp/vendor"}], indirect=True)
+def test_metadata_azure(
+    copy_files: list[Path],
+    chalk: Chalk,
+    tmp_file: IO,
+    server_imds: str,
+):
+    # make imds plugin think we are running in EC2
+    with tmp_file as fid:
+        fid.write(b"Microsoft Corporation")
+    bin_path = copy_files[0]
+    insert = chalk.insert(bin_path, config=CONFIGS / "imds.conf")
+    assert insert.report.contains(
+        {
+            "_OP_CLOUD_PROVIDER": "azure",
+            "_AZURE_INSTANCE_METADATA": {
+                "compute": {
+                    "azEnvironment": "AzurePublicCloud",
+                    "customData": "",
+                    "evictionPolicy": "",
+                    "isHostCompatibilityLayerVm": "true",
+                    "licenseType": "",
+                    "location": "westeurope",
+                    "name": "myVm",
+                    "offer": "0001-com-ubuntu-server-focal",
+                    "osProfile": {
+                        "adminUsername": "testuser",
+                        "computerName": "myVm",
+                        "disablePasswordAuthentication": "true",
+                    },
+                    "osType": "Linux",
+                    "placementGroupId": "",
+                    "plan": {"name": "", "product": "", "publisher": ""},
+                    "platformFaultDomain": "0",
+                    "platformUpdateDomain": "0",
+                    "priority": "",
+                    "provider": "Microsoft.Compute",
+                    "publicKeys": [
+                        {
+                            "keyData": "ssh-rsa AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJQPr4RsDbaJdKPHl2gfCwiWcTRVEu0XlQvsPgdvCH/Io8Im1VfBMamtRhTIEqlEoTaRD8h9ETDQAPg7GUVkg07P3ZgDfFf94KePpxADso7GoqaPsGuL4OQpURa4DQCmf1Jw+kDg0TI1ERYIQoNOGduiS5cuB74A5BxcgW2A52ocVoiINS1tPudZBIvnr8iQXa6BhB5EgUVP0w+pGaOgI4jHga8ThT9weGqzBrtBcyiZ44jfT2Tg/AjI4GuXq14HdFEN0096vk= generated-by-azure",
```

Comment:
there is no need to hardcode full RSA key in tests. you can match the value by regex:

```python
"keyData": re.compile(r"...."),
```

see aws example above how to do that

Reply:
yeah its a dummy public key anyhow so it was easier to simply paste